x-pack/plugins/cloud_security_posture/public/pages/compliance_dashboard/compliance_dashboard.tsx

```diff
@@ -239,12 +243,15 @@ export const ComplianceDashboard = () => {
         label: i18n.translate('xpack.csp.dashboardTabs.cloudTab.tabTitle', {
           defaultMessage: 'Cloud',
         }),
+        'data-test-subj': CLOUD_DASHBOARD_TAB,
         isSelected: selectedTab === CSPM_POLICY_TEMPLATE,
         onClick: () => setSelectedTab(CSPM_POLICY_TEMPLATE),
         content: (
           <>
-            {hasFindingsCspm ? (
-              <CloudPosturePage query={getCspmDashboardData}>
+            {hasFindingsCspm || !isCspmInstalled ? (
+              <CloudPosturePage
+                query={getCspmDashboardData.data ? getCspmDashboardData : undefined}
+              >
```

Comment:
the point of the `query` property is to get a query and handle its loading and error states. this change makes it handle only the data state, skipping loading and error.

can you elaborate from your research what exactly caused the problem this pr is fixing and maybe we can look for a solution that doesn't require changing the way the current properties are meant to be used?

Reply:
so the issue we have here originally (which is what causing the 1st issue in the 1st place) is that we run either getKspmDashboardData or getCspmDashboardData but we have no data, the isLoading field kept returning true (never goes to false) as such we get stuck on the loading page because of it